ddtrace/internal/telemetry/data/payload.py

```diff
@@ -0,0 +1,133 @@
+from abc import ABC
+from abc import abstractmethod
+from typing import Dict
+from typing import List
+
+from ....version import get_version
+from .dependency import Dependency
+from .dependency import create_dependency
+from .integration import Integration
+from .metrics import Series
+
+
+class Payload(ABC):
+    """ "
+    Meta class which ensures all telemetry payloads implement
+    the abstract methods listed below.
+    """
+
+    @abstractmethod
+    def request_type(self):
+        # type: () -> str
+        """
+        Every payload must return one of the following request types:
+        app-closed, app-started, app-dendencies-load, app-integrations-changed,
+        app-heartbeat, app-generate-metrics
+        """
+        pass
+
+    @abstractmethod
+    def to_dict(self):
+        # type: () -> Dict
+        """
+        The return value of this method is used to convert a Payload object
+        into a json. All payload fields that are required by the telemetry intake
+        service must be set here
+        """
+        pass
```

Comment:
We should `raise NotImplementedError` in both of these.

`abc.abstractmethod` can still be called.